homeassistant/components/tplink_omada/config_flow.py

```diff
@@ -0,0 +1,199 @@
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+
+        errors: dict[str, str] = {}
+        info = None
+        if user_input is not None:
+            info = await self._test_login(user_input, errors)
+
+        if info is None or user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
+            )
+
+        await self.async_set_unique_id(info.controller_id)
+        self._abort_if_unique_id_configured()
+
+        self._omada_opts.update(user_input)
+        self._sites = info.sites
+        self._controller_name = info.name
+        if len(self._sites) > 1:
+            return await self.async_step_site()
+        return await self.async_step_site({CONF_SITE: self._sites[0].id})
+
+    async def async_step_site(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle step to select site to manage."""
+
+        if user_input is None:
+            schema = vol.Schema(
+                {
+                    vol.Required(CONF_SITE, "site"): selector.SelectSelector(
+                        selector.SelectSelectorConfig(
+                            options=[
+                                selector.SelectOptionDict(value=s.id, label=s.name)
+                                for s in self._sites
+                            ],
+                            multiple=False,
+                            mode=selector.SelectSelectorMode.DROPDOWN,
+                        )
+                    )
+                }
+            )
+
+            return self.async_show_form(step_id="site", data_schema=schema)
+
+        self._omada_opts.update(user_input)
+        site_name = next(
+            site for site in self._sites if site.id == user_input["site"]
+        ).name
+        display_name = f"{self._controller_name} ({site_name})"
+
+        return self.async_create_entry(title=display_name, data=self._omada_opts)
+
+    async def async_step_reauth(self, entry_data: Mapping[str, Any]) -> FlowResult:
+        """Perform reauth upon an API authentication error."""
+        self._omada_opts = dict(entry_data)
+        return await self.async_step_reauth_confirm(dict(entry_data))
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Dialog that informs the user that reauth is required."""
+
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            self._omada_opts.update(user_input)
+            info = await self._test_login(self._omada_opts, errors)
+
+            if info is not None:
+                # Auth successful - update the config entry with the new credentials
+                entry = self.hass.config_entries.async_get_entry(
+                    self.context["entry_id"]
+                )
+                if entry is not None:
```

Comment:
You can use `assert entry is not None` to satisfy typing. If you really expect the entry to be gone (it's theoretically possible I guess), you should add an `else: return self.async_abort(…)`

Reply:
Yeah, I think it's fine to assume `async_get_entry` will get something back, so this was just to shut up the null warning. I'm still learning python features one at a time.